backend/internal/storage/postgres/works.go

```diff
@@ -0,0 +1,180 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/CamPlume1/khoury-classroom/internal/models"
+	"github.com/jackc/pgx/v5"
+)
+
+// squashes a list of student work contributors to a list of student works with an array of contributors
+func formatWorks(rawWorks []models.StudentWork) []models.FormattedStudentWork {
+	// map to group works by AssignmentOutlineID
+	workMap := make(map[int]*models.FormattedStudentWork)
+
+	for _, work := range rawWorks {
+		if _, exists := workMap[work.ID]; !exists {
+			// insert new formatted work if doesn't exist already
+			workMap[work.ID] = &models.FormattedStudentWork{
+				StudentWork:  work,
+				Contributors: []string{},
+			}
+		}
+
+		// combine first and last names into a full name and add to list of contributors
+		fullName := fmt.Sprintf("%s %s", work.FirstName, work.LastName)
+		workMap[work.ID].Contributors = append(workMap[work.ID].Contributors, fullName)
+	}
+
+	// convert map values to a slice
+	var formattedWorks []models.FormattedStudentWork
+	for _, formattedWork := range workMap {
+		formattedWorks = append(formattedWorks, *formattedWork)
+	}
+
+	return formattedWorks
+}
+
+// squashes a list of student work contributors to a list of student works with an array of contributors
+func formatPaginatedWorks(rawWorks []models.PaginatedStudentWork) []models.FormattedPaginatedStudentWork {
+	// map to group works by AssignmentOutlineID
+	workMap := make(map[int]*models.FormattedPaginatedStudentWork)
+
+	for _, work := range rawWorks {
+		if _, exists := workMap[work.ID]; !exists {
+			// insert new formatted work if doesn't exist already
+			workMap[work.ID] = &models.FormattedPaginatedStudentWork{
+				PaginatedStudentWork: work,
+				Contributors:         []string{},
+			}
+		}
+
+		// combine first and last names into a full name and add to list of contributors
+		fullName := fmt.Sprintf("%s %s", work.FirstName, work.LastName)
+		workMap[work.ID].Contributors = append(workMap[work.ID].Contributors, fullName)
+	}
+
+	// convert map values to a slice
+	var formattedWorks []models.FormattedPaginatedStudentWork
+	for _, formattedWork := range workMap {
+		formattedWorks = append(formattedWorks, *formattedWork)
+	}
+
+	return formattedWorks
+}
```

Comment:
done with some interfaces, still a bit ugly but i think that's as good as it can get with the data type variety that comes with our data formatting needs